src/styles/annotator/_utilities.scss

```diff
@@ -0,0 +1,3 @@
+@use '@hypothesis/frontend-shared/styles/util';
+
+@use 'tailwindcss/utilities';
```

Comment:
Is the order important here? Which of these two sets of styles should take precedence if there is a conflict, or should there never be a conflict?

Reply:
Not to give away the ending, but the use of the `frontend-shared` utility styles will be going away entirely within the next few PRs. We're transitioning to only using tailwind-generated classes for _utilities_. At this _time_ there should be no conflict in the cascade as shown here.

To answer part of your actual question: as we go here, we'll be making tighter use of tailwind's layers. Tailwind's utility classes may make up our entire utility layer, or we may inject a few classes into the utilities layer, e.g.:

```scss
@tailwind utilities; // This is equivalent to @use 'tailwindcss/utilities' but is more idiomatic to TW

@layer utilities {
  .u-whatever {
    @apply border-pink;
  }
}
```

I haven't yet encountered a specific use case in our apps for the above, but it is certainly possible. If we wanted to override, say, a color or a border width or something _for the entire app_ versus the upstream preset, we can do that in the local tailwind config.

Finally, probably most strategically relevantly, I'm working on an approach to put the styles for shared components into tailwind's `components` layer instead of using the `utilities` layer (which is what happens de-facto if one uses TW utilities within component JSX). I haven't quite figured out the mechanics of that, but I have some sketchy ideas in my head. By putting shared-component styles in TW's components layer, we can feel confident that when apps add additional classes to shared components, e.g.:

```jsx
<SomeContainer classes="text-lg">My stuff<SomeContainer>
```

If `SomeContainer`'s styling sets a text size also, but within the `components` layer, we can feel confident that the above use will be guaranteed of overriding it because the `text-lg` is on the `utilities` layer.